OpenRA.Game/Game.cs

```diff
@@ -232,6 +232,21 @@ public static void RestartGame()
 			// Reseed the RNG so this isn't an exact repeat of the last game
 			lobbyInfo.GlobalSettings.RandomSeed = CosmeticRandom.Next();
 
+			// Update MapCache
+			if (ModData.MapCache[lobbyInfo.GlobalSettings.Map].Status != MapStatus.Available)
```

Comment:
This check doesn't appear to be needed (and isn't used in `LobbyLogic`)?

Reply:
it is used when you restart the game, its part of the code that fixes the #19988 crash. It both updates MapCache and checks if the selected map is not outdated

Reply:
I mean that we can run the body of this unconditionally, since `GetUpdatedMap` will also update the `MapCache` and return the original UID unchanged if it still exists.

Reply:
you are correct, I'll remove the check